src/sentry/static/sentry/app/utils/queryString.tsx

```diff
@@ -33,13 +33,17 @@ type QueryValue = string | string[] | undefined | null;
 export function appendTagCondition(
   query: QueryValue,
   key: string,
-  value: string
+  value: string,
+  wrap_with_tag?: boolean
```

Comment:
When is this false?

Reply:
I wanted to have minimal impact - it is false in every case except the one where this is invoked on the issue details page, where I explicitly set it to true. I felt I didn't understand enough about the implications of this work to make a riskier decision. Would have to do much testing if I changed this to ALWAYS wrap.

Reply:
I feel like mostly we want to use the old syntax. The newer syntax is mostly a workaround for when we have field clashes, so this seems correct? Or is this only used for when people click on a tag?

Reply:
It looks like `tag[xxx]` format would be enforced. Frontend would need to be aware which tags are preserved.

Reply:
This is why using the `query` param from the serializer is a good idea. It means the backend can decide how to render the query, and the frontend doesn't need any special logic.

Reply:
I agree, we should use the query attribute, but I saw the front end forming the query using key:value and not the query attribute. This may be a bigger change (and a different PR). This happens in eventTags.tsx line 46.

Reply:
Also seems relevant:
Should the syntax be `tag[x]:y` or `tags[x]:y`?

Reply:
I vote for `tag[x]:y`.